index.js

```diff
@@ -127,25 +145,20 @@ const drawCommand = (message) => {
   });
 };
 
-/*
-const inventoryCommand = () => {
-
-if user draws a card
-store name of card to firebase under the key cardName
-when the same user draws a second card, store the name of the second card under the previous draw within cardName
-continue to do this to all additional cards
-
-user{
-  userID{
-    lastDrawTime: xxxx
-    username: ""
-    cardName: "Blue Eyes White Dragon", "Red Eyes Black Dragon"
-  }
-}
+const inventoryCommand = (message) => {
+  var ref = db.ref("users");
+  ref.once("value", function (snapshot) {
+    const userData = snapshot.val();
+    const userIdentification = message.author.id;
+    const userInventory = userData[userIdentification].inventory;
+    let cardData = Object.values(userInventory);
+    const inventoryList = [];
 
+    cardData.forEach((item) => inventoryList.push("•  " + item.name));
```

Comment:
`.forEach` has served us well for the past hour, but it's a littleeee bit more generic than we need. There's an even more purpose-built ES6 array function that is made for what we're doing. It's called `.map` and is probably the most used array function. We used it a lot in the frontend code for the pokedb.

A `arrayName.map()` takes an array and returns an array. Take a look at the MDN docs for it -- it may feel hard to grasp and different from things you've done before, but it will become easy and second nature.

What you want to do is eliminate the need for a `.push` at all -- instead you'll initialize your array of message lines to equal exactly what you need.

e.g. your code will look like `const inventoryList = cardData.map( etc );` and eliminate the `forEach` and `.push`.


Not to make this more overwhelming, but it actually is a different _paradigm_ in programming, called "functional programming", as opposed to the normal "imperative programming" that you're used to. Schools teach imperative, but the world is moving towards programming more functionally. Don't pay too much attention to trying to learn the difference now, it'll be something we get into more over time.